src/com/tony/server/Authenticator.java

```diff
@@ -0,0 +1,56 @@
+package com.tony.server;
+
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+
+public class Authenticator {
+    private static final int ENCODED_USER_INFO = 1;
+    private ArrayList<Request> authenticationList;
+    private ArrayList<String> authenticatedUsers;
+
+    public Authenticator() {
+        this.authenticationList = new ArrayList<>();
+        this.authenticatedUsers = new ArrayList<>();
+    }
+
+    public void addToAuthenticationList(Request request) {
+       authenticationList.add(request);
+    }
+
+    public void addToAuthenticatedUsers(String userInfo) {
+        authenticatedUsers.add(userInfo);
+    }
+
+    public ArrayList<Request> getAuthenticationList() {
+        return authenticationList;
+    }
+
+    public ArrayList<String> getAuthenticatedUsers() {
+        return authenticatedUsers;
+    }
+
+    public boolean requiresAuthentication(Request request) {
+        return authenticationList.contains(request);
+    }
+
+    public boolean isAuthenticated(Request request) {
+        HashMap headers = request.getHeaders();
+        if(headers.containsKey("Authorization")){
+            String authorizationValue = request.getHeaders().get("Authorization");
```

Comment:
Why do you pull out the `headers` on line 38, only use them in the `if` statement on line 39, then fetch the `headers` again on line 40?